source_code/main_mcu/src/fido2/fido2.c

```diff
@@ -0,0 +1,493 @@
+    memcpy(response->aaguid, MINIBLE_AAGUID, AAGUID_LEN);
+    //We no longer have the credential ID. Not needed for assertion
+    response->cred_ID_len = 0;
+    response->flags = auth_data_header.flags;
+    //error code already set
+    return response->error_code;
+}
+
+/*! \fn     fido2_process_make_auth_data(FIDO2_make_auth_data_req_message_t const *request, FIDO2_make_auth_data_rsp_message_t *response)
+*   \brief  Make the authentication data. This esentially creates the key pair and stores
+*           the new record in the DB
+*   \param  incoming request messsage
+*   \param  outgoing response message
+*   \return void
+*/
+void fido2_process_make_auth_data(FIDO2_make_auth_data_req_message_t const *request, FIDO2_make_auth_data_rsp_message_t *response)
+{
+    memset(response, 0, sizeof(*response));
+
+    if (request->new_cred == 1)
+    {
+        fido2_make_auth_data_new_cred(request, response);
+    }
+    else
+    {
+        fido2_make_auth_data_existing_cred(request, response);
+    }
+}
+
+/*! \fn     fido2_process_get_next_credential(FIDO2_get_next_credential_req_message_t const *request, FIDO2_get_next_credential_rsp_message_t *response)
+*   \brief  Process getting the next credential for an rpID
+*   \param  incoming request messsage
+*   \param  outgoing response message
+*   \return void
+*/
+void fido2_process_get_next_credential(FIDO2_get_next_credential_req_message_t const *request, FIDO2_get_next_credential_rsp_message_t *response)
+{
+    credential_DB_rec_t *rec = NULL;
+    memset(response, 0, sizeof(*response));
+
+    rec = fido2_get_credential_by_index(request->rpID, request->index);
+
+    if (rec != NULL)
+    {
+        //Fill out response object
+        memcpy(response->tag, rec->tag, TAG_LEN);
+        memcpy(response->user_ID, rec->user_ID, USER_ID_LEN);
+        response->result = 1;
+    }
+    else
+    {
+        response->result = 0;
+    }
+}
+
+//DEBUG STORAGE:
+#define MAX_CRED 10
+static uint8_t cred_count = 0;
+static credential_DB_rec_t cred_store[MAX_CRED];
+
+static bool fido2_room_for_more_creds(uint8_t const *rpID)
+{
+    //TODO: Look up in DB
+    return cred_count < MAX_CRED;
+}
+
+static void fido2_store_credential_in_db(credential_DB_rec_t const *cred_rec)
+{
+    //TODO: Debug impl
+    if (fido2_room_for_more_creds(cred_rec->rpID))
+    {
+        memcpy(&cred_store[cred_count++], cred_rec, sizeof(cred_store[0]));
+    }
+}
+
+static void fido2_update_credential_in_db(credential_DB_rec_t const *cred_rec)
+{
+    //TODO: Update flash
+}
+
+static credential_DB_rec_t *fido2_get_credential(uint8_t const *rpID, uint8_t const *tag)
+{
+    //TODO: Debug impl
+    uint32_t i;
+
+    for(i = 0; i < MAX_CRED; ++i)
+    {
+        if (memcmp(cred_store[i].tag, tag, TAG_LEN) == 0)
+        {
+            return &cred_store[i];
+        }
+    }
+    return NULL;
+}
+
+static credential_DB_rec_t *fido2_get_credential_by_index(uint8_t const *rpID, uint32_t index)
+{
```

Comment:
not needed?

Reply:
All of this is for debug until we have something stored in DB. I assume you will remove all these debug ones when you implement the storage....